feed/utils.js

```diff
@@ -0,0 +1,50 @@
+async function getActivityFeedData(query = {}, nextLink) {
+  let finalUrl =
+    API_BASE_URL + (nextLink || '/logs' + generateActivityFeedParams(query));
+  const res = await fetch(finalUrl, {
+    credentials: 'include',
+    method: 'GET',
+    headers: {
+      'Content-type': 'application/json',
+    },
+  });
+  return await res.json();
+}
+
+function generateActivityFeedParams(query) {
+  let queryParam = 'dev=true&format=feed';
+  if (query?.category) queryParam += getLogTypesFromCategory(query?.category);
+  return `?${queryParam}`;
+}
+
+function getLogTypesFromCategory(category) {
+  switch (category) {
+    case CATEGORY.ALL:
+      return '';
+    case CATEGORY.TASK:
+      return `&type=${logType.TASK}`;
+    case CATEGORY.EXTENSION_REQUESTS:
+      return `&type=${logType.EXTENSION_REQUESTS}`;
+    case CATEGORY.TASK_REQUESTS:
+      return `&type=${logType.TASK_REQUESTS}`;
+    case CATEGORY.OOO:
+      return `&type=${logType.REQUEST_CREATED},${logType.REQUEST_REJECTED},${logType.REQUEST_APPROVED}`;
+  }
+}
+
+function addEmptyPageMessage(container) {
+  const emptyPageMessage = createElement({
+    type: 'p',
+    attributes: { class: 'page-message' },
+    innerText: 'No logs to show!',
+  });
+  container.appendChild(emptyPageMessage);
+}
+
+function addErrorElement(container) {
```

Comment:
Consider providing a more informative message to the user. This could include:
A generic message like "An error occurred while fetching data. Please try again later."
If possible, include a specific error code or message for developers or debugging purposes.

Reply:
<img width="706" alt="image" src="https://github.com/Real-Dev-Squad/website-dashboard/assets/61227144/51874698-8dea-46f2-970b-cb5d7a95ec36">
If error will occur this will be shown to the user, this works?